File: back_traking/Queen_permutation.java
package back_traking;

import java.util.*;

class Queen_permute {
    public static int count = 0;
    public static void queenPermutations(boolean boxes[], int qpsf, int tq, String ans) {

        if(qpsf == tq){
            count++;
            System.out.println(ans);
            return;
        }

        for (int i = 0; i < boxes.length; i++) {
            if (boxes[i] == false) {
                boxes[i] = true;
                queenPermutations(boxes, qpsf + 1, tq, ans + "q" + qpsf + "b" + i + " ");
                boxes[i] = false;  //undo step.
            }
        }
    }

    public static void main(String[] args) {
        boolean[] boxes = new boolean[4];
        queenPermutations(boxes, 0, 2, "");
        System.out.println(count);
    }
} 

File: back_traking/Queen_permutation.py
def queenPermute(boxes, qpsf, tq, ans):
    global count
    if qpsf == tq:
        count += 1
        print(ans)
        return

    for i in range(len(boxes)):
        if boxes[i] == False:
            boxes[i] = True
            queenPermute(boxes, qpsf + 1, tq, ans + 'q' + str(qpsf) + 'b' + str(i) + " ")
            boxes[i] = False

count = 0
boxes = [False] * 4
queenPermute(boxes, 0, 2, '')
print(count)